File: Tests/Parsing/LepersColonyPageScrapingTests.m
//  LepersColonyPageScrapingTests.m
//
//  Copyright 2013 Awful Contributors. CC BY-NC-SA 3.0 US https://github.com/Awful/Awful.app

#import "AwfulParsingTestCase.h"
#import "AwfulLepersColonyPageScraper.h"

@interface LepersColonyPageScrapingTests : AwfulParsingTestCase

@end

@implementation LepersColonyPageScrapingTests

+ (Class)scraperClass
{
    return [AwfulLepersColonyPageScraper class];
}

- (void)testFirstPage
{
    NSArray *bans = [self scrapeFixtureNamed:@"banlist"];
    XCTAssertEqual(bans.count, (NSUInteger)50);
    NSArray *allUsers = [AwfulUser fetchAllInManagedObjectContext:self.managedObjectContext];
    XCTAssertEqual(allUsers.count, (NSUInteger)71);
    NSArray *allPosts = [AwfulPost fetchAllInManagedObjectContext:self.managedObjectContext];
    XCTAssertEqual(allPosts.count, (NSUInteger)46);
    
    AwfulBan *first = bans[0];
    XCTAssertEqual(first.punishment, AwfulPunishmentProbation);
    XCTAssertEqualObjects(first.post.postID, @"421665753");
    XCTAssertEqual(first.date.timeIntervalSince1970, 1384078200.);
    XCTAssertEqualObjects(first.user.username, @"Kheldragar");
    XCTAssertEqualObjects(first.user.userID, @"202925");
    XCTAssertNotEqual([first.reasonHTML rangeOfString:@"shitty as you"].location, NSNotFound);
    XCTAssertEqualObjects(first.requester.username, @"Ralp");
    XCTAssertEqualObjects(first.requester.userID, @"61644");
    XCTAssertEqualObjects(first.approver, first.requester);
}

@end

File: Core/Tests/LepersColonyPageScrapingTests.swift
//  LepersColonyPageScrapingTests.swift
//
//  Copyright 2013 Awful Contributors. CC BY-NC-SA 3.0 US https://github.com/Awful/Awful.app

@testable import AwfulCore
import XCTest

final class LepersColonyPageScrapingTests: XCTestCase {
    override class func setUp() {
        super.setUp()

        makeUTCDefaultTimeZone()
    }

    func testFirstPage() {
        let result = try! scrapeFixture(named: "banlist") as LepersColonyScrapeResult
        XCTAssertEqual(result.punishments.count, 50)
        
        let first = result.punishments[0]
        XCTAssertEqual(first.sentence, .probation)
        XCTAssertEqual(first.post?.rawValue, "421665753")
        XCTAssertEqual(first.date?.timeIntervalSince1970, 1384078200)
        XCTAssertEqual(first.subjectUsername, "Kheldragar")
        XCTAssertEqual(first.subject?.rawValue, "202925")
        XCTAssert(first.reason.contains("shitty as you"))
        XCTAssertEqual(first.requesterUsername, "Ralp")
        XCTAssertEqual(first.requester?.rawValue, "61644")
        XCTAssertEqual(first.approver, first.requester)
        XCTAssertEqual(first.approverUsername, first.requesterUsername)
    }
}
